blobfuse/src/OAuthTokenCredentialManager.cpp

```diff
@@ -194,36 +185,14 @@ OAuthToken OAuthTokenCredentialManager::refresh_token()
 /// </summary>
 OAuthToken OAuthTokenCredentialManager::get_token()
 {
-    #ifdef TOKEN_REFRESH_THREAD
-    if ((!valid_authentication) && is_token_expired()) {
-        // Lock the mutex.
-        if (!token_mutex.try_lock()) {
-            fprintf(stdout, "Locking mutex failed, so some token is being acquired., so just wait and get that\n");
-            syslog(LOG_WARNING, "OAUTH Token : Locking mutex failed, so some token is being acquired., so just wait and get that\n");
-
-            time_t current_time;
-            current_time =  get_current_time_in_utc();
-
-            // There's a five minute segment where the token hasn't actually expired yet, but we're already trying to refresh it.
-            // We can just use the currently active token instead, rather than waiting for the refresh.
-            // This'll save some downtime on systems under constant use.
-            if (current_time < current_oauth_token.expires_on) {
-                syslog(LOG_DEBUG, "OAUTH Token : Locking mutex failed. buffer time is there use old token\n");
-                return current_oauth_token;
-            } else {
-                // If it's not still live, let's just wait for the refresh to finish.
-                // This is a sub-optimal method to wait for this event as it can end up blocking other routines after the token has finished refreshing.
-                // If we were working in Go, I (Adele) would suggest using a sync.WaitGroup for this functionality.
-                syslog(LOG_ERR, "OAUTH Token : Wait for token to get refreshed as buffer time has expired");
-                token_mutex.lock();
-            }
-        }
-
-        // Note that we don't always lock in this function and sometimes return early.
-        // Be sure to ensure you're safely manipulating the lock when modifying this function.
-        token_mutex.unlock();
+    #ifndef TOKEN_REFRESH_THREAD
+    // since the thread would have updated the expired token just check if valid_authentication
+   // and return the current token, control will fall to the if is_token_expired() below if this directive is undefined.
+    if (valid_authentication)
+    {
+        return current_oauth_token;
     }
-    #else
+    #endif

```

Comment:
No need to call refresh_token() if the thread is running, we just need to wait for the thread to refresh the token.

Reply:
if valid_authentication has failed the thread has gone wrong so refreshing the token here even if the #def TOKEN_REFRESH_THREAD isdefined

Reply:
understood the logic.